sdk/analytics/src/main/java/com/klaviyo/analytics/UserInfo.kt

```diff
@@ -97,4 +131,23 @@ internal object UserInfo {
             profile.setPhoneNumber(phoneNumber)
         }
     }
+
+    fun shouldUpdatePush(
+        pushToken: String,
+        permission: Boolean? = null,
+        background: Boolean? = null
+    ): Boolean =
+        pushState != formatPushState(pushToken, permission, background)
+
+    fun setPushToken(token: String, permission: Boolean? = null, background: Boolean? = null) {
+        pushToken = token
+        pushState = formatPushState(pushToken, permission, background)
+    }
+
+    private fun formatPushState(
+        token: String,
+        permission: Boolean? = null,
+        background: Boolean? = null
+    ): String =
+        "$token-${permission ?: notificationPermission}-${background ?: backgroundData}"
```

Comment:
Sorry am I overlooking something, but where is `notificationPermission` getting set?

Reply:
Oh yeah that reads funny, without context. `notificationPermission` is an imported member off the DeviceProperties object. A computed var that gets permission setting from the device (background is the same)